_posts/2022-08-31-quarkus-helm.adoc

```diff
@@ -0,0 +1,251 @@
+author: jcarvaja
+---
+
+https://helm.sh/[Helm] is the most popular package manager that finds, shares, and deploys software built for Kubernetes. To configure Java applications for distribution through Helm, you need to package them as Helm charts containing all of the necessary metadata and resource definitions and this can be a very tedious task. The good news is that Quarkus has recently released the new https://github.com/quarkiverse/quarkus-helm[Quarkus Helm] extension that will automatically generate all these Helm chart manifests and resources!
+
+== Introduction to Helm charts
+
+First of all, let's introduce how a Helm chart looks like. A Helm chart is a collection of files inside a directory. The following files are mandatory:
+
+- `Chart.yaml`: Chart metadata, such as name, version, and developers.
+- `values.yaml`: Default configuration values for the chart in YAML.
+- `templates`: A directory containing the list of resources that, in combination with values.yaml, generate the application when the chart is installed. The resources are also specified in YAML.
+
+Find more information about Helm charts in the https://helm.sh/docs/topics/charts/[official Helm documentation].
+
+== Getting started with the Quarkus Helm extension
+
+Let's now see how we can generate the Helm chart of a Quarkus application. 
+
+In this example, we’ll create a Quarkus application with the Quarkus Helm and the Quarkus Kubernetes extensions by running the following command:
+
+[source, bash]
+----
+mvn io.quarkus.platform:quarkus-maven-plugin:2.12.0.Final:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=helm-quickstart \
+    -DclassName="org.acme.quickstart.GreetingResource" \
+    -Dpath="/hello" \
+    -Dextensions="resteasy-reactive,kubernetes,helm"
+cd helm-quickstart
+----
+
+If you already have your Quarkus project configured, you can add the helm extension to your project by running the following command in your project base directory:
+
+[source, bash]
+----
+./mvnw quarkus:add-extension -Dextensions="helm"
+----
+
+This command will add the following dependency to your `pom.xml` file:
+
+[source, xml]
+----
+<dependency>
+    <groupId>io.quarkiverse.helm</groupId>
+    <artifactId>quarkus-helm</artifactId>
+</dependency>
+----
+
+And now you can generate the Helm resources using the Maven build command:
+
+[source, bash]
+----
+./mvnw clean package
+----
+
+Depending on the Quarkus Kubernetes extensions that you are using in your project, the Helm resources will include some resources or others. Since we have used the Quarkus Kubernetes extension, the Helm chart resources will be generated in the `target/helm/kubernetes/<chart name>/` directory. 
+
+[TIP]
+====
+The `<chart name>` is set from either the property `quarkus.helm.name` or the Quarkus application.
+====
+
+Moreover, the Helm resources include the following files in the `target/classes/META-INF/dekorate/helm/<chart name>/templates` directory:
```

Comment:
So the templates are included in the jar?

Reply:
No, wrong path, fixing it.